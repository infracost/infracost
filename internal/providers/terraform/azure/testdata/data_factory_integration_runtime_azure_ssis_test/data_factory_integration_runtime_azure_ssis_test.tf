provider "azurerm" {
  skip_provider_registration = true
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_data_factory" "example" {
  name                = "example"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
}

resource "azurerm_data_factory_integration_runtime_azure_ssis" "default" {
  name            = "runtime-default"
  data_factory_id = azurerm_data_factory.example.id
  location        = azurerm_resource_group.example.location

  node_size = "Standard_D8_v3"
}

resource "azurerm_data_factory_integration_runtime_azure_ssis" "standard_ahb_v2" {
  name            = "runtime-standard"
  data_factory_id = azurerm_data_factory.example.id
  location        = azurerm_resource_group.example.location

  node_size       = "Standard_D1_v2"
  number_of_nodes = 4
  edition         = "Standard"
  license_type    = "BasePrice"
}

resource "azurerm_data_factory_integration_runtime_azure_ssis" "enterprise_with_license" {
  name            = "runtime-enterprise"
  data_factory_id = azurerm_data_factory.example.id
  location        = azurerm_resource_group.example.location

  node_size    = "Standard_E64_v3"
  edition      = "Enterprise"
  license_type = "LicenseIncluded"
}
