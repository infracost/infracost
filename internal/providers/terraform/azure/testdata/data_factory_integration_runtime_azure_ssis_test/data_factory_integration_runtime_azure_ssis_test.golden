
 Name                                                                               Monthly Qty  Unit                    Monthly Cost 
                                                                                                                                      
 azurerm_data_factory.example                                                                                                         
 ├─ Read/Write operations                                                     Monthly cost depends on usage: $0.50 per 50k entities   
 └─ Monitoring operations                                                     Monthly cost depends on usage: $0.25 per 50k entities   
                                                                                                                                      
 azurerm_data_factory_integration_runtime_azure_ssis.default                                                                          
 └─ Compute (D8 v3, Standard, license included)                                             730  hours                      $1,414.74 
                                                                                                                                      
 azurerm_data_factory_integration_runtime_azure_ssis.enterprise_with_license                                                          
 └─ Compute (E64 v3, Enterprise, license included)                                          730  hours                     $25,821.56 
                                                                                                                                      
 azurerm_data_factory_integration_runtime_azure_ssis.standard_ahb_v2                                                                  
 └─ Compute (D1 v2, Standard)                                                             2,920  hours                        $588.53 
                                                                                                                                      
 OVERALL TOTAL                                                                                                             $27,824.83 
──────────────────────────────────
5 cloud resources were detected:
∙ 4 were estimated, 1 of which usage-based costs, see https://infracost.io/usage-file
∙ 1 was free:
  ∙ 1 x azurerm_resource_group